### PR TITLE
Communicate switch status to semantics on Windows

### PR DESCRIPTION
**Pull request by yaakovschectman**
Switch widgets used to have the `PUSHBUTTON` role on Windows, which has no associated state. This PR switches it to a `CHECKBUTTON` and associates the `PRESSED` state with its MSAA node when it is toggled. Adds a unit test to verify that the native state reflects the toggle state of the Switch.

Also fixes some typos in comments in the existing checkbox unit test.

Addresses [#flutter/flutter/111525](https://github.com/flutter/flutter/issues/111525)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/flutter_windows_view_unittests.cc [hunk 1]
```diff
@@ -716,13 +716,104 @@ TEST(FlutterWindowsViewTest, CheckboxNativeState) {
...
+
+    // Look up against the node itself (not one of its children).
+    VARIANT varchild = {};
+    varchild.vt = VT_I4;
+
+    // Verify the switch is not pressed.
+    varchild.lVal = CHILDID_SELF;
+    VARIANT native_state = {};
+    ASSERT_TRUE(SUCCEEDED(native_view->get_accState(varchild, &native_state)));
+    EXPECT_FALSE(native_state.lVal & STATE_SYSTEM_PRESSED);
```
Comment (chunhtai) [inline shell/platform/windows/flutter_windows_view_unittests.cc:814]: Is it possible to also test its MSAARole is  ROLE_SYSTEM_CHECKBUTTON?

File: third_party/accessibility/ax/platform/ax_platform_node_win.cc [hunk 2]
```diff
@@ -3421,7 +3421,7 @@ int AXPlatformNodeWin::MSAARole() {
       return ROLE_SYSTEM_TITLEBAR;
 
     case ax::mojom::Role::kToggleButton:
-      return ROLE_SYSTEM_PUSHBUTTON;
+      return ROLE_SYSTEM_CHECKBUTTON;
```
Comment (chunhtai) [inline third_party/accessibility/ax/platform/ax_platform_node_win.cc:3424]: Is check button basically a check box? Why do we choose this over push button?

Comment (yaakovschectman) [reply third_party/accessibility/ax/platform/ax_platform_node_win.cc:3424]: It is the same role as for a checkbox. In [#flutter/fluter/111525](https://github.com/flutter/flutter/issues/111525) we are currently of the opinion that it is the most appropriate role, as it is for interactable widgets with toggleable state, and the [materials design guidelines](https://material.io/components/switches#usage) recommend their use in place of checkboxes on mobile

Comment (chunhtai) [reply third_party/accessibility/ax/platform/ax_platform_node_win.cc:3424]: thanks for explanation. SGTM

**Review by chunhtai — APPROVED** [review]
LGTM

**Review by loic-sharma — APPROVED** [review]
LGTM - nice work!

**Pull request merged**